### DISCUSSION TITLE
AI tasked with "Recharge Battery Cells" replace Fulgurium Batteries at any charge level below 100% instead of 25%

### DISCUSSION THREAD

**Discussion by Next-Door-Tech** [FakeFishGames/Barotrauma#14449] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

AI tasked with "Recharge Battery Cells" will take Fulgurium Batteries out of tools at any non-full charge percentage (i.e. 99% or lower), as opposed to regular batteries which are only replaced at or below 25% (I believe). This can be rather annoying since equipment that uses batteries may be stored in a different locker than the preferred location for extra battery cells, meaning that extra trips around the ship are needed when preparing for EVA, for instance.

### Reproduction steps

1. Task an AI crew member with recharging battery cells.
2. Put a fulgurium battery into any applicable tool, and use it once, so the battery is no longer 100% full.
3. Place the tool into any container accesible to the AI tasked with recharging.
4. The AI will remove the fulgurium battery from the tool and place it in the ship's charging dock, even if it is still at 99% charge.

If the procedure is repeated with a regular battery cell instead, step 4 only occurs when the battery is below 25% charge.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

MacOS and Windows

**Comment by Regalis11** [comment]
Thank you for the report! Fixed in https://github.com/FakeFishGames/Barotrauma-development/commit/d086c10542e74a6cafc303b7645d10d25d4042ca (commit in our private repo), fix coming up in the next update.
